dowhy/gcm/anomaly.py

```diff
@@ -0,0 +1,186 @@
+from typing import Callable, Optional, Any, Union, Dict
+
+import numpy as np
+import pandas as pd
+from numpy.matlib import repmat
+from tqdm import tqdm
+
+from dowhy.gcm import config
+from dowhy.gcm._noise import compute_noise_from_data, noise_samples_of_ancestors, get_noise_dependent_function
+from dowhy.gcm.anomaly_scorer import AnomalyScorer
+from dowhy.gcm.anomaly_scorers import MedianCDFQuantileScorer, RescaledMedianCDFQuantileScorer
+from dowhy.gcm.cms import ProbabilisticCausalModel, InvertibleStructuralCausalModel
+from dowhy.gcm.graph import ConditionalStochasticModel, get_ordered_predecessors, is_root_node, validate_causal_dag
+from dowhy.gcm.shapley import ShapleyConfig, estimate_shapley_values
+from dowhy.gcm.stats import permute_features
+from dowhy.gcm.util.general import shape_into_2d
+
+
+def estimate_conditional_anomaly_score(parent_samples: np.ndarray,
+                                       target_samples: np.ndarray,
+                                       causal_model: ConditionalStochasticModel,
+                                       anomaly_scorer_factory: Callable[[], AnomalyScorer] = MedianCDFQuantileScorer,
+                                       num_samples_conditional: int = 10000) -> np.ndarray:
+    """ Estimates the conditional anomaly score based on the expected outcomes of the causal model. """
+    parent_samples, target_samples = shape_into_2d(parent_samples, target_samples)
```

Comment:
are these samples from all the parents or a single parent?

Reply:
From all parents. I tried to clarify this by extending the docstring.